doc-bridgepoint/notes/9706_polys/9706_polys_dnt.md

```diff
@@ -0,0 +1,244 @@
+set through modifying the `plugin_customization.ini` file. The default delivered
+with the tool is that concrete polymorphic events are _disabled_.
+
+6.2 UI elements and changes
+
+6.2.1 Model changes
+
+The changes associated with UI elements were done almost completely in OAL in
+the OOA of OOA. Logic was implemented in key places to implement the behaviors
+analyzed in section 5. Checking of the preference was done as sparingly as
+possible. The state of the model was used whenever possible to determine
+"concreteness" (e.g. if a non-local event was pointing to a polymorphic event in
+the same state machine, it is assumed to be a concrete poly. There is no need to
+check the preference here).
+
+6.2.2 Non-model changes
+
+The model explorer archetypes and PEI were changed to prevent non-local events
+from being displayed in the tree if the polymorphic event is in the same state
+machine.
+
+The State Event Matrix editor was modified to not show a column for polymorphic
+events. This was actually a change from current behavior event with abstract
+polys, but it was decided that there is no use for showing these columns.
+
+6.3 Architecture modifications
+
+6.3.1 MC-Java
+
+MC-Java was extended to generate selection methods for this general case:
+```
+select one ... related by ... where ...
+```
+Before, generating such a statement in MC-Java would result in Java that would
+not compile. This enhancement was needed for this implementation.
+
+6.3.2 Persistence
+
+The stream exporter uses PEI data to export proxy elements for other elements.
+This is used in copy and paste. Conventionally, non-local events need a proxy to
+the state machine instances of the event they refer to. However, when the
+non-local event is in the same state machine as the polymorphic event, exporting
+these instances can result in an infinite recursion. The PEI and archetypes
+needed to be updated to handle this new case.
+
+### 7. Design Comments
+
+None
+
+### 8. User Documentation
+
+8.1 The documentation that described the concrete polymorphic event naming
+idiom shall be removed
+
+8.2 Documentation of the new project preference to enable concrete polymorphic
+events shall be introduced
+
+### 9. Unit Test
+
+9.1 MASL round trip shall pass
+
+9.2 Editing test  
+9.2.1 Before opening BridgePoint:  
+9.2.1.1 Open the file `plugin_customization.ini` which can be found under the
+`plugins/plugins/org.xtuml.bp.pkg_<build_id>/` directory.  
+9.2.1.2 Find the line that says:  
+```
+org.xtuml.bp.core/bridgepoint_prefs_allow_concrete_polys=false
+```
+9.2.1.3 Change the value to `true`  
+9.2.2 Open a new workspace. Assure that the default action language dialect is
+"MASL".  
+9.2.3 Convert and import the model at `masl/test/poly` in the `models` repo
+using the MASL conversion utility.  
+9.2.4 Create a new project in the workspace.
+9.2.5 Attempt to duplicate the `poly` model. Take note of any errors logged or
+strange behavior.  
+
```

Comment:
Seems like there should be some steps added:
* verify what the user should see in model explorer
* make sure the poly event can be assigned in a subtype
* make sure it can't be deleted in the subtype

Unless by step 9.2.5 you mean to recreate the poly model in a new project by hand...

Reply:
Yes, by hand. the purpose of this test is to verify the editing capabilities of BridgePoint around the area of concrete polys. It's hard to specify _exactly_ what the user should see I think.